### PR TITLE
Add common retry helper for SeverManager

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Comment by pakrym** [comment]
VSTS https://github.com/aspnet/IISIntegration/issues/1049

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs [hunk 1]
```diff
@@ -178,115 +178,45 @@ private async Task WaitUntilSiteStarted()
...
 
-                throw new TimeoutException($"IIS failed to stop site {site}.");
-            }
+                HostProcess = Process.GetProcessById(workerProcess.ProcessId);
+                Logger.LogInformation("Site has started.");
+            });
         }
 
-        private static bool IsExpectedException(Exception ex)
+        public void StopAndDeleteAppPool()
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs]: Remove this method?

Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs]: Yep

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs [hunk 2]
```diff
@@ -378,14 +293,96 @@ private void ConfigureAppHostConfig(XElement config, string contentRoot, int por
...
+            try
             {
-                var site = serverManager.Sites.SingleOrDefault();
-                site.Stop();
-                var appPool = serverManager.ApplicationPools.SingleOrDefault();
-                appPool.Stop();
-                serverManager.CommitChanges();
+                RetryServerManagerAction(serverManager =>
+                {
+                    var appPool = serverManager.ApplicationPools.SingleOrDefault();
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs]: Shouldn't appPool the order of site and appPool be swapped? Stop the site before stopping the app pool?

Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployer.cs]: Good point.

**Pull request merged**